### PR TITLE
XW-3672 | Power mobile recirc unit with LiftIgniter

### PR DESCRIPTION
**Pull request by JamesHawking**
## Links

* https://wikia-inc.atlassian.net/browse/XW-3672

## Description

AC:
* Recommendations are powered by liftigniter
* Verify that Sponsored Content correctly appears Options

## Reviewers

@Wikia/x-wing

### CODE REVIEW

File: app/services/liftigniter.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import Ember from 'ember';
+
+const {Service, $, inject} = Ember;
+const localStorageAdapter = require('mobile-wiki/utils/local-storage-connector').localStorageAdapter;
```
Comment (rogatty) [inline app/services/liftigniter.js]: Use `import localStorageConnector from '../utils/local-storage-connector';`

File: app/inline-scripts/tracking-liftigniter.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+	w[v] = w[v] || function () {
+			(w[v].q = w[v].q || []).push(
+				arguments)
+		};
+	w[v].l = 1 * new Date();
+	e = d.createElement(s), r = d.getElementsByTagName(s)[0];
+	e.async = 1;
+	e.src = p + '?ts=' + (+new Date() / 3600000 | 0);
+	r.parentNode.insertBefore(e, r)
+})(window, document, 'script', '//cdn.fandommetrics.com/l9ehhrb6mtv75bp2.js', '$p');
```
Comment (rogatty) [inline app/inline-scripts/tracking-liftigniter.js]: Can we use `liftIgniter` instead of `$p`?

File: app/index.html [hunk 3]
```diff
@@ -10,6 +10,7 @@
 {{content-for 'tracking-comscore'}}
 {{content-for 'tracking-ua-init'}}
 {{content-for 'head-footer'}}
+{{content-for 'tracking-liftigniter'}}
```
Comment (rogatty) [inline app/index.html]: We keep the business-critical scripts here. Can we move this just above `{{content-for 'body-footer'}}`?

File: app/components/recirculation-prefooter.js [hunk 4]
```diff
@@ -1,25 +1,53 @@
 import Ember from 'ember';
 import InViewportMixin from 'ember-in-viewport';
-import FandomPostsModel from '../models/fandom-posts';
 import {track, trackActions} from '../utils/track';
 
-const {Component, getOwner, on, run} = Ember;
+const {Component, on, run, inject, $} = Ember;
+const recircItemsCount = 10;
```
Comment (rogatty) [inline app/components/recirculation-prefooter.js]: Let's put it inside the component or move config here. For consistency.

File: app/services/liftigniter.js [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+				globalCtx: context,
+			}
+		});
+
+		window.liftigniter('send', 'pageview');
+		window.liftigniter('setRequestFields', ['rank', 'thumbnail', 'title', 'url', 'presented_by', 'author']);
+	},
+
+	getData(config) {
+		const deferred = $.Deferred(),
```
Comment (rogatty) [inline app/services/liftigniter.js]: Let's use Ember's `RSVP.defer()`, see https://github.com/tildeio/rsvp.js/#deferred

File: app/styles/component/_recirculation.scss [hunk 6]
```diff
@@ -4,6 +4,44 @@
 
   &__item {
 	min-height: 150px;
+	position: relative;
+
+	&--anchor {
```
Comment (rogatty) [inline app/styles/component/_recirculation.scss]: We don't use the double dash.

File: app/styles/component/_recirculation.scss [hunk 7]
```diff
@@ -4,6 +4,44 @@
...
+	  color: $wds-color-black;
+	}
+
+	&--is-video {
```
Comment (rogatty) [inline app/styles/component/_recirculation.scss]: `&.is-video`?

File: app/templates/components/recirculation-prefooter.hbs [hunk 8]
```diff
@@ -1,10 +1,23 @@
...
-	<h3>{{model.title}}</h3>
+{{#if items.length}}
+	<h3>{{title}}</h3>
 	<ul class="recirculation-prefooter__list">
-		{{#each model.posts as |post index|}}
+		{{#each items as |post index|}}
 			<li class="recirculation-prefooter__item" {{action 'postClick' post index preventDefault=true}}>
-				<img src="{{post.image_url}}" alt="{{post.title}}">
-				<p class="recirculation-prefooter__title">{{post.title}}</p>
+				<!--This 'a' tag is required by liftigniter's js lib in order to properly set tracking for elements-->
```
Comment (rogatty) [inline app/templates/components/recirculation-prefooter.hbs:6]: `a` or `div`?

File: app/components/recirculation-prefooter.js [hunk 9]
```diff
@@ -1,25 +1,51 @@
...
 		didEnterViewport() {
-			const fandomPosts = FandomPostsModel.create(getOwner(this).ownerInjection());
+			const liftigniter = this.get('liftigniter');
+
+			liftigniter
+				.getData(config)
+				.done((data) => {
+					this.setProperties({
+						isVisible: true,
+						title: this.get('i18n').t('recirculation.trending-fandom-articles'),
```
Comment (rogatty) [inline app/components/recirculation-prefooter.js]: Why not use it directly in the template?

File: app/templates/components/recirculation-prefooter.hbs (hunk 8 above)
Comment (rybmat) [reply app/templates/components/recirculation-prefooter.hbs:6]: `a`, `div` is not needed here at all

File: app/services/liftigniter.js [hunk 10]
```diff
@@ -0,0 +1,112 @@
...
+				globalCtx: context,
+			}
+		});
+
+		window.liftigniter('send', 'pageview');
+		window.liftigniter('setRequestFields', ['rank', 'thumbnail', 'title', 'url', 'presented_by', 'author']);
+	},
+
+	getData(config) {
+		const deferred = RSVP.defer(),
```
Comment (hakubo) [inline app/services/liftigniter.js:72]: more common way of ding this is
return RSVP.Promise((resolve, reject) {
...
resolve(response)
})

but I would not change it now,

File: app/styles/component/_recirculation.scss [hunk 11]
```diff
@@ -4,13 +4,52 @@
 
   &__item {
 	min-height: 150px;
+	position: relative;
+
+	&-anchor {
+	  color: $wds-color-black;
```
Comment (hakubo) [inline app/styles/component/_recirculation.scss]: spaces

**Review by hakubo — CHANGES_REQUESTED** [review]

**Review by hakubo — APPROVED** [review]

**Review by rogatty — APPROVED** [review]

**Pull request merged**